docs/edgeql/select.rst

```diff
@@ -1026,6 +1026,70 @@ exclusively fetch the ``Movie.characters`` of type ``Hero``.
     ...
   }
 
+Accessing types in polymorphic queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the type of an object is displayed inside polymorphic queries, this
+is simply a convenience of the REPL and not a property that can be accessed.
+This is particularly noticeable if you cast an object to ``json``, making it
+impossible to determine the type if the query is polymorphic:
+
+.. code-block:: edgeql-repl
+
+    db> select Person limit 1;
+    {default::Villain {id: 6c22bdf0-5c03-11ee-99ff-dfaea4d947ce}}
+    db> select <json>Person limit 1;
+    {Json("{\"id\": \"6c22bdf0-5c03-11ee-99ff-dfaea4d947ce\"}")}
+
+The type of an object is found inside ``__type__`` which is a link that
+carries various information about a type, including its ``name``.
```

Comment:
```suggestion
carries various information about the object's type, including its ``name``.
```
I always fear there could be confusion between data type and object type, so I like to be explicit about which we're talking about.